client/alloc_runner.go

```diff
@@ -643,10 +644,18 @@ func (r *AllocRunner) setTaskState(taskName, state string, event *structs.TaskEv
 		r.appendTaskEvent(taskState, event)
 	}
 
-	if state == "" {
+	if lazySync {
```

Comment:
Why do we need a new flag for this? Can't we skip sync'ing if new state == old state (meaning just a local event has occurred)?